pkg/types/ccip/onramp.go

```diff
@@ -59,4 +59,14 @@ type OnRampReader interface {
 	Address() (Address, error)
 
 	GetDynamicConfig() (OnRampDynamicConfig, error)
+
+	// IsSourceChainHealthy returns true if the source chain is healthy.
+	IsSourceChainHealthy(ctx context.Context) (bool, error)
+
+	// IsSourceCursed returns true if the source chain is cursed. OnRamp communicates with the underlying RMN
+	// to verify if source chain was cursed or not.
+	IsSourceCursed(ctx context.Context) (bool, error)
```

Comment:
Why is RMN communication chain-specific?

Reply:
There's one RMN per chain